RedHat-UX/red-hat-design-system
# fix(footer): match sidegap to the design specs

## What I did

1. Moved the sidegap spacing rules from the responsive.css sheet to the the lightdom.css file.
2. Added three levels of spacing changes based on breakpoint. 
 - default: (--rh-space-lg, 16px)
 - (--rh-breakpoint-sm, 768px): (--rh-space-2xl, 32px)
 - (~~--rh-breakpoint-lg, 1200px~~ --rh-breakpoint-xl, 1440): (--rh-space-4xl, 64px)
3. Introduced a **global custom property** called `--rhds-section-side-gap` that will be the single source of truth for elements that need to keep their content aligned with the footer content.


## Testing Instructions

1. Go here: https://deploy-preview-841--red-hat-design-system.netlify.app/elements/footer/demo/
2. Make sure that the space between the viewport and the content matches the [design specs](https://xd.adobe.com/view/835616bd-1374-483d-ab10-6ae92e0e343c-d605/screen/4c0466ca-93a2-4846-a0b5-07378eeadc87).

## Notes to Reviewers

**Why did you move it to the lightdom?**

There was a flash of unstyled content that would be a bit jarring.

**What's with this global variable --rhds-section-side-gap?**

Other sites are going to struggle lining up their headers with our footer if we don't expose our business logic for these three levels of spacing.  The alternative would be that every change to our internal sidebar spacing styles would in essence be a breaking change.

I'm not sure how I feel about this yet though so I would love your input!

**Are you positive about the breakpoint values?**

No, not really.  I would appreciate a design review to gut check me.


## Screenshots

![Screenshot 2023-03-20 at 9 02 19 PM](https://user-images.githubusercontent.com/3428964/226497672-47f52c8f-f19f-4bff-8155-0b607aebbf35.png)

![Screenshot 2023-03-20 at 9 02 07 PM](https://user-images.githubusercontent.com/3428964/226497696-fae5f220-a1eb-4d2b-8706-bc71c739b236.png)

![Screenshot 2023-03-23 at 9 08 54 PM](https://user-images.githubusercontent.com/3428964/227398750-7fb48bd3-d6f7-4eed-b6a6-5546133cf376.png)
